src/operations/blas3/gemm_local.hpp

```diff
@@ -565,23 +652,30 @@ class Gemm<input_t, output_t, DoubleBuffer, NbcA, NbcB, ClSize, TileType,
    * @param reg_b  temporary register used to prefetch elements of B
    * @param reg_res  2D register array used to store the result C
    */
-  template <typename InputPointerType>
-  SYCL_BLAS_INLINE void compute_block_gemm(InputPointerType B,
+  template <bool check_m_limit, bool check_n_limit, typename InputPointerType>
+  SYCL_BLAS_INLINE void compute_block_gemm(index_t item_id, InputPointerType B,
                                            InputPointerType A, element_t *reg_a,
                                            element_t &reg_b,
                                            element_t *reg_res) noexcept {
-    // NOTE: Adding "#pragma unroll" here reduces performance on AMD R9 Nano.
-    //       Seems that the small reduction of arithmetic operations does not
-    //       amortize the cost of loading the larger kernel binary resulting
-    //       from loop unrollment.
+    // NOTE: Adding "#pragma unroll" here reduces performance on AMD R9
+    // Nano.
+    //       Seems that the small reduction of arithmetic operations does
+    //       not amortize the cost of loading the larger kernel binary
+    //       resulting from loop unrollment.
+    constexpr index_t work_per_load =
+        !check_m_limit && !check_n_limit ? packetize_t::packet_size : 1;
     for (index_t i = 0; i < cl_elems; ++i) {
 #pragma unroll
-      for (index_t j = 0; j < item_rows; ++j) {
-        reg_a[j] = A[j * wg_rows];
+      for (index_t j = 0; j < item_rows / work_per_load; ++j) {
+#pragma unroll
+        for (int i = 0; i < work_per_load; i++) {
+          reg_a[i + j * work_per_load] =
+              *(A + (i + j * wg_rows * work_per_load));
+        }
       }
 #pragma unroll
       for (index_t j = 0; j < item_cols; ++j) {
-        reg_b = B[j * ldsb];
+        reg_b = *(B + j * ldsb);
```

Comment:
the is no difference between the two the compiler does the same for the former.

Reply:
This avoids a bunch of compiler warnings as the kernel uses multi_ptrs and the use of subscript with them is deprecated.